### DISCUSSION TITLE
2 questions about classes/order/OrderSlip.php

### DISCUSSION THREAD

**Discussion by BenTen** [PrestaShop/PrestaShop#33752] [Q&A]
When we look at the method `create()` in https://github.com/PrestaShop/PrestaShop/blob/8.1.x/classes/order/OrderSlip.php

- _bool_ **shipping_cost**, default `true`
- _float_ **amount**, default `0`
- _bool_ **amount_choosen**, default `false`
- _bool_ **add_tax**, default `true`

The attribut **shipping_cost_amount** will take the value of **total_shipping_tax_incl** depends of **shipping_cost** et **add_tax**.

By default **total_shipping_tax_incl** take the value 0 until **shipping_cost** become `true`

When **shipping_cost** become `true`

```php

        if ($shipping_cost !== false) {
            $order_slip->shipping_cost = true;
            $carrier = new Carrier((int) $order->id_carrier);
            $address = Address::initialize($order->id_address_delivery, false);
            $tax_calculator = $carrier->getTaxCalculator($address);
            $order_slip->{'total_shipping_tax_' . $inc_or_ex_1} = ($shipping_cost === null ? $order->{'total_shipping_tax_' . $inc_or_ex_1} : (float) $shipping_cost);

            if ($tax_calculator instanceof TaxCalculator) {
                $order_slip->{'total_shipping_tax_' . $inc_or_ex_2} = Tools::ps_round($tax_calculator->{$taxCalculatorMethod}($order_slip->{'total_shipping_tax_' . $inc_or_ex_1}), Context::getContext()->getComputingPrecision());
            } else {
                $order_slip->{'total_shipping_tax_' . $inc_or_ex_2} = $order_slip->{'total_shipping_tax_' . $inc_or_ex_1};
            }
        } else {
            $order_slip->shipping_cost = false;
        }
```

Here we try to check the carrier and address to find calculation of taxes.

And  then we check **add_tax**.

If the value is `false`

```php
$order_slip->{'total_shipping_tax_' . $inc_or_ex_1} = ($shipping_cost === null ? $order->{'total_shipping_tax_' . $inc_or_ex_1} : (float) $shipping_cost);
```

So finally, my question is :

`$shipping_cost === null` 
shipping_cost is boolean, is it correct to compare to `null` with `===`?

`: (float) $shipping_cost);`
**shipping_cost** is a bool, why do we try to convert in float?

**Comment by matks** [comment]
Hi @BenTen I'm afraid this method was created 8 years ago according to the git blame 😅 https://github.com/PrestaShop/PrestaShop/blame/8.1.x/classes/order/OrderSlip.php#L261 so I'm not confident that someone will be able to answer
